app/experimenter/experiments/tests/serializers/test_timeline_population.py

```diff
@@ -0,0 +1,122 @@
+        locale = LocaleFactory.create()
+        serializer = LocaleSerializerMultiSelect(locale)
+
+        self.assertEqual(serializer.data, {"value": locale.code, "label": locale.name})
+
+
+class TestCountrySerializerMultiSelect(serializers.ModelSerializer):
+
+    def test_serializer_outputs_expected_schema(self):
+        country = CountryFactory.create()
+        serializer = CountrySerializerMultiSelect(country)
+
+        self.assertEqual(serializer.data, {"value": country.code, "label": country.name})
+
+
+class TestExperimentTimelinePopSerializer(MockRequestMixin, TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        self.experiment = ExperimentFactory.create(
+            type=ExperimentConstants.TYPE_PREF,
+            locales=[LocaleFactory.create()],
+            countries=[CountryFactory.create()],
+        )
+
+    def test_serializer_outputs_expected_schema(self):
+
+        serializer = ExperimentTimelinePopSerializer(self.experiment)
+
+        self.assertEqual(
+            serializer.data["proposed_start_date"],
+            self.experiment.proposed_start_date.strftime("%Y-%m-%d"),
+        )
+        self.assertEqual(
+            serializer.data["firefox_channel"], self.experiment.firefox_channel
+        )
+        self.assertEqual(serializer.data["platform"], self.experiment.platform)
+
+    def test_all_values_are_optional(self):
+        data = {}
+
+        serializer = ExperimentTimelinePopSerializer(instance=self.experiment, data=data)
+
+        self.assertTrue(serializer.is_valid())
+
+    def test_serializer_saves_values(self):
+        country = CountryFactory()
+        locale = LocaleFactory()
+        countries = [{"value": country.id, "label": country.name}]
+        locales = [{"value": locale.id, "label": locale.name}]
+
+        data = {
+            "firefox_min_version": "67.0",
+            "firefox_max_version": "68.0",
+            "countries": countries,
+            "locales": locales,
+        }
+
+        serializer = ExperimentTimelinePopSerializer(
+            instance=self.experiment, data=data, context={"request": self.request}
+        )
+
+        self.assertTrue(serializer.is_valid())
+
+        experiment = serializer.save()
+
+        self.assertEqual(experiment.changes.count(), 1)
+        self.assertEqual(experiment.firefox_min_version, "67.0")
+        self.assertEqual(experiment.firefox_max_version, "68.0")
+
+    def test_serializer_rejects_firefox_min_less_max(self):
+        data = {
+            "firefox_min_version": "69.0",
+            "firefox_max_version": "68.0",
+            "countries": [],
+            "locales": [],
+        }
+
+        serializer = ExperimentTimelinePopSerializer(instance=self.experiment, data=data)
+
+        self.assertFalse(serializer.is_valid())
+
+    def test_serializer_rejects_date_in_past(self):
+        data = {"proposed_start_date": "2019-2-01", "countries": [], "locales": []}
```

Comment:
Don't change this, that'll always pass (unless the server falls into a time warp), but in general for testing dates like this I usually do something like `datetime.date.today() - datetime.timedelta(wahtever)`